chain/ethereum/src/network_indexer/ethereum.graphql

```diff
@@ -0,0 +1,86 @@
+""" Block is an Ethereum block."""
+type Block @entity {
+  id: ID!
+
+  """The number of this block, starting at 0 for the genesis block."""
+  number: BigInt!
+
+  """The block hash of this block."""
+  hash: Bytes!
+
+  """The parent block of this block."""
+  parent: Block
```

Comment:
This is based on https://github.com/ethereum/go-ethereum/blob/d90d1db609c8d77baa422d49bd371207c06b4711/graphql/schema.go#L146-L147. Note that having a parent hash and having a parent block that _exists_ are two different things. Let's say genesis has `0x0...0` as its parent hash. That doesn't mean it has an actual parent.

Reply:
We could store both the `parentHash` and the parent reference. What do you think?

Reply:
The only block where that would be interesting is the genesis block though, I think. So it may be a lot of overhead for nothing. Because in all other cases, we can get to the parent block hash via `parent { hash }`.

Reply:
I see, yhea this really should be nullable.